<?xml version="1.0"?>

<robot name="hybrid_robotv0_2" xmlns:xacro="http://ros.org/wiki/xacro">

  <xacro:include filename="$(find hybrid_jumping_robot)/urdf/material.xacro"/>
  <xacro:include filename="$(find hybrid_jumping_robot)/urdf/plugins.gazebo"/>
  <xacro:include filename="$(find hybrid_jumping_robot)/urdf/macro.xacro"/>
  <xacro:include filename="$(find hybrid_jumping_robot)/urdf/transmission.xacro"/>

  <xacro:property name="wheel_r" value="0.15"/>
  <xacro:property name="wheel_width" value="0.02"/>
  <xacro:property name="base_length" value="1"/>
  <xacro:property name="base_width" value="0.5"/>
  <xacro:property name="base_height" value="0.1"/>
  <xacro:property name="extension_length" value="1"/>
  <xacro:property name="extension_width" value="0.45"/>
  <xacro:property name="extension_height" value="0.095"/>

  <link name="base">
      <visual>
          <geometry>
              <box size="${base_length} ${base_width}  ${base_height}"/>
          </geometry>
          <material name="blue"/>
      </visual>
      <collision>
          <geometry>
              <box size="${base_length} ${base_width}  ${base_height}"/>
          </geometry>
      </collision>
      <inertial>
          <mass value="10"/>
          <inertia ixx="0.21666666666666667" ixy="0" ixz="0" iyy="0.8416666666666667" iyz="0" izz="1.0416666666666665"/>
      </inertial>
  </link>

  <link name="extension">
      <visual>
          <geometry>
              <box size="${extension_length} ${extension_width} ${extension_height}"/>
          </geometry>
          <material name="red"/>
      </visual>
      <collision>
          <geometry>
              <box size="${extension_length} ${extension_width} ${extension_height}"/>
          </geometry>
      </collision>
      <inertial>
          <mass value="10"/>
          <inertia ixx="0.17627083333333335" ixy="0" ixz="0" iyy="0.8408541666666667" iyz="0" izz="1.0020833333333334"/>
      </inertial>
  </link>
  <joint name="extension_base_joint" type="prismatic">
      <axis xyz="1 0 0"/>
      <parent link="base"/>
      <child link="extension"/>
      <origin xyz="-0.1 0 0"/>
      <limit lower="-0.9" upper="0" effort="10000" velocity="1000"/>
      <joint_properties damping="1.0" friction="1.0" />
  </joint>

  <xacro:wheel position="front" side="right" parent="base" wheel_r="${wheel_r}" wheel_width="${wheel_width}"  base_length="${base_length}" base_width="${base_width}"/>
  <xacro:wheel position="front" side="left" parent="base" wheel_r="${wheel_r}" wheel_width="${wheel_width}"  base_length="${base_length}" base_width="${base_width}"/>
  <xacro:wheel position="back" side="right" parent="extension" wheel_r="${wheel_r}" wheel_width="${wheel_width}"  base_length="${base_length}" base_width="${base_width}"/>
  <xacro:wheel position="back" side="left" parent="extension" wheel_r="${wheel_r}" wheel_width="${wheel_width}"  base_length="${base_length}" base_width="${base_width}"/>

</robot>
